staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -123,9 +123,9 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx context.Context, obj runtime.
 		objectMeta.SetManagedFields(nil)
 	}
 
-	// ClusterName is ignored and should not be saved
-	if len(objectMeta.GetClusterName()) > 0 {
-		objectMeta.SetClusterName("")
+	// ZZZ_DeprecatedClusterName is ignored and should not be saved
+	if len(objectMeta.GetZZZ_DeprecatedClusterName()) > 0 {
+		objectMeta.SetZZZ_DeprecatedClusterName("")
```

Comment:
issue a warning to the requester here, e.g.:

```go
warning.AddWarning(ctx, "", "metadata.clusterName was specified, but is not preserved and the API field will be removed in 1.25")
```

Reply:
done